**Question A (a)** The comparison "Biden vs. Trump" is more representative because it combines aspects of past U.S. presidential election performance with future predictions of the winner, and we have four years of data for Biden and Trump as U.S. presidents to compare. However, in the case of "Obama vs. Bush", we are not only evaluating past performance, but also need to predict future performance, for which there is a lack of complete data and Biden has withdrawn from the election.

**Question A (b)**
The differences in the data sample are even more pronounced when considering the Biden-Trump comparison, because we can use data from their previous terms to form assumptions about their performance. This is in contrast to the original “Hillary vs. Trump” comparison, when neither candidate had presidential data to draw on. The Biden-Harris scenario extends the message of this data even further.

**Question A What is hypothesis?**
A hypothesis is a statement or assumption about a population parameter. Hypothesis testing compares the data collected from a sample with the null hypothesis to determine if there is enough evidence to reject the null hypothesis while supporting the alternative hypothesis.

**Question A**
Chat Histories https://chatgpt.com/share/670efd0f-ace8-800f-bd83-abad74d209a8

**Question 1**
In hypothesis testing, the null hypothesis (H₀) assumes that there is no significant effect, difference, or relationship in the research object, and the alternative hypothesis (H₁) assumes that there is an effect or difference. The goal of hypothesis testing is to determine whether there is enough evidence to reject the null hypothesis in favor of the alternative hypothesis through data analysis. Testable ideas need to meet two key conditions: one is measurable, and the other is falsifiable, which means that the idea must be able to be quantitatively evaluated through data and it is possible to prove it wrong by analyzing the data. A good null hypothesis should be clear and specific, and assume a state of "no effect" or "no difference". When designing the null hypothesis, make sure it can be tested through statistical data, so that it is scientifically determined whether the hypothesis can be rejected and the research conclusions can be drawn.

Chat Histories https://chatgpt.com/share/670eff0b-851c-800f-8749-dd4fe6096236

**Question 2**
In statistical testing, we typically draw a sample from a larger population and analyze the sample data to infer the characteristics of the population. The data points in the sample (like \( x_i \)) are used to calculate sample statistics, such as the sample mean μ , which helps us estimate the parameters of the population. The true mean of the population μ0 is the main focus of interest, but we usually do not know its exact value.
Therefore, the primary goal of statistical tests is to use information from the sample to infer whether the population parameter is consistent with our assumptions. For example, in hypothesis testing, we compare the sample data to the hypothesized population mean μ0 to determine whether the sample provides sufficient evidence to reject the null hypothesis. This means that while we analyze the sample data, the real focus is on making inferences about the population’s true characteristics, not just drawing conclusions about the sample itself.
Thus, the outcome of statistical tests is centered on the population parameter rather than the sample statistic. This process ensures that we can make reasonable inferences about a larger population or phenomenon based on limited sample data.



**Question 3**
When calculating the p-value, we start our study from this point assuming that the null hypothesis is true. The null hypothesis states that there is no effect or difference in the population being studied, and that any differences in this set of studies are due to random chance.
By imagining a world where the null hypothesis is true, we can investigate the likelihood that the data we collected (or more extreme data) could have been observed by chance, and the p-value tells us this likelihood. If the p-value is very small, it indicates that it is unlikely that the observed data occurred by chance, providing us with evidence against the null hypothesis.

Chat Histories https://chatgpt.com/share/670f033c-43f4-800f-a178-51601d1cea79

**Question 4**
As the p-value decreases, it indicates that the data extreme observed under the null hypothesis is less likely to occur, which makes the null hypothesis seem less plausible. This is because the p-value represents the probability of obtaining such an extreme result when the null hypothesis is true. Therefore, the smaller the p-value, the stronger the evidence against the null hypothesis, making it seem more "ridiculous" or unlikely to be true.

Chat Histories https://chatgpt.com/share/670f053b-c954-800f-8c22-5b07b009b239

In [2]:
#Question 5
import numpy as np

# Step 1: Define the parameters
n_couples = 124             # Total number of couples
observed_right_tilts = 80    # Observed number of right tilts
num_simulations = 10000      # Number of simulations

# Step 2: Perform the simulation
print("Performing simulation...")
simulations = np.random.binomial(n=n_couples, p=0.5, size=num_simulations)
print("Simulation complete.")

# Step 3: Calculate the p-value
print("Calculating p-value...")
p_value = np.mean(simulations >= observed_right_tilts)
print(f"P-value: {p_value:.5f}")

# Step 4: Determine the level of evidence using the provided table
if p_value > 0.1:
    evidence = "No evidence against the null hypothesis"
elif 0.1 >= p_value > 0.05:
    evidence = "Weak evidence against the null hypothesis"
elif 0.05 >= p_value > 0.01:
    evidence = "Moderate evidence against the null hypothesis"
elif 0.01 >= p_value > 0.001:
    evidence = "Strong evidence against the null hypothesis"
else:
    evidence = "Very strong evidence against the null hypothesis"

# Step 5: Display the evidence
print(f"Evidence: {evidence}")


Performing simulation...
Simulation complete.
Calculating p-value...
P-value: 0.00070
Evidence: Very strong evidence against the null hypothesis


**Question 5**
Based on the provided table, the p-value from the simulation is 0.00070, which falls into the category:
0.001≥p
According to the table, this corresponds to **"Very strong evidence against the null hypothesis."**

Thus, the simulation provides very strong evidence against the null hypothesis that couples have no preference for tilting their heads to the right or left when kissing.

**Question 6**
Even if the p-value is small, we still cannot definitively prove that the null hypothesis is false. The p-value simply measures the strength of the evidence against the null hypothesis; the lower the value, the stronger the evidence. However, it does not guarantee that the null hypothesis is false. Statistical hypothesis testing is based on probability and uncertainty, not absolute evidence.

**Question 6**
Chat Histories https://chatgpt.com/share/670f0f18-e4e8-800f-a7aa-8039c954049a

In [4]:
#Question 7
import pandas as pd
import numpy as np

# Simulate a dataset
np.random.seed(42)  # Ensure reproducibility
n_patients = 10  # Let's assume we have data for 10 patients
health_score_change = np.random.normal(loc=0, scale=1, size=n_patients)  # Random normal changes

# Create the DataFrame
patient_data = pd.DataFrame({'HealthScoreChange': health_score_change})

# Display the simulated data
print(patient_data)

# Compute observed proportion of patients with improvement
observed_proportion = (patient_data['HealthScoreChange'] > 0).mean()

# Simulate random improvements under the null hypothesis
np.random.seed(1)  # Make the simulation reproducible
number_of_simulations = 10000
IncreaseProportionSimulations_underH0random = np.zeros(number_of_simulations)

for i in range(number_of_simulations):
    random_improvement = np.random.choice([0, 1], size=len(patient_data), replace=True)
    IncreaseProportionSimulations_underH0random[i] = random_improvement.mean()

# One-sided p-value (right-tailed)
p_value = (IncreaseProportionSimulations_underH0random >= observed_proportion).mean()

print(f"One-sided p-value: {p_value}")



   HealthScoreChange
0           0.496714
1          -0.138264
2           0.647689
3           1.523030
4          -0.234153
5          -0.234137
6           1.579213
7           0.767435
8          -0.469474
9           0.542560
One-sided p-value: 0.373


**Question 7**
Several changes were made to the code:
A simulated dataset of changes in health scores for 10 patients was added, and the observed proportion of patients who improved was calculated: this is a key statistic for comparison with the null distribution.
Assuming the null hypothesis (improvement by random chance), 10,000 simulations were run to create a distribution of proportions of improvement under random chance.

**Question 7**
Chat Histories https://chatgpt.com/share/670f122f-2de8-800f-b1c8-95bf44bac870

In [5]:
#Question 8
import numpy as np
from scipy.stats import norm

# Set a random seed for reproducibility
np.random.seed(42)

# Given values
p_hat = 49 / 80  # observed sample proportion
p_0 = 0.5  # null hypothesis proportion
n = 80  # sample size

# Calculate z-statistic
z_stat = (p_hat - p_0) / np.sqrt(p_0 * (1 - p_0) / n)

# Calculate p-value (one-tailed test)
p_value = 1 - norm.cdf(z_stat)

print("z-statistic:", z_stat)
print("p-value:", p_value)


z-statistic: 2.0124611797498115
p-value: 0.022085672454221217


**Question 8**
The null hypothesis (H₀) assumes that STA130 students are merely guessing, with a 50% chance of correctly identifying whether tea or milk was poured first (\(p = 0.5\)). The alternative hypothesis (H₁) posits that students have some ability to correctly identify the order (\(p > 0.5\)). A z-test for proportions was conducted, comparing the observed sample proportion of 0.6125 to the hypothesized value of 0.5. The resulting z-statistic was 2.01, and the p-value was 0.0221. Since the p-value is less than 0.05, we reject H₀, indicating that students' performance is likely better than random guessing.

**Question 8**
https://chatgpt.com/share/670f1411-ba34-800f-b49c-ece94306d86d

**Question 9**
I have reviewed the course wiki-textbook and interacted with a ChatBot to help me to understand all the material in the tutorial and lecture that I didn't quite follow when I first saw it.